modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminDisplayContext.java

```diff
@@ -752,13 +768,25 @@ private SearchContainer<RepositoryEntry> _getDLSearchContainer()
 
 			_initializeFilterSearchContext(searchContext);
 
-			Indexer<?> indexer = IndexerRegistryUtil.getIndexer(
+			List<RepositoryEntry> repositoryEntries = new ArrayList<>();
+
+			Indexer<?> dlFileEntryIndexer = IndexerRegistryUtil.getIndexer(
 				DLFileEntryConstants.getClassName());
 
-			Hits hits = indexer.search(searchContext);
+			repositoryEntries.addAll(
+				_getRepositoryEntries(
+					dlFileEntryIndexer.search(searchContext)));
+
+			if (isOnlyNavigationFilter()) {
+				Indexer<?> dlFolderIndexer = IndexerRegistryUtil.getIndexer(
+					DLFolderConstants.getClassName());
+
+				repositoryEntries.addAll(
+					_getSearchResults(dlFolderIndexer.search(searchContext)));
+			}
```

Comment:
Isn't this going to break pagination? In the worst case, pageSize*2 elements will be returned.

Maybe we need to take an approach similar to the ContentDashboard? See com.liferay.content.dashboard.web.internal.portlet.action.GetContentDashboardItemsXlsMVCResourceCommand#_getSearchResponse. Instead of using the `Indexer`s, we may need to use the `Searcher` directly; that way we can ask for elements of multiple classNames (FileEntry and Folder in this case).

Reply:
Sorry, you're completely right. I will close this pull and revisit the changes when time allows. Thank you for the feedback.